payment/service.go

```diff
@@ -0,0 +1,158 @@
+package payment
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/brave-intl/bat-go/utils/clients/cbr"
+	"github.com/brave-intl/bat-go/wallet/provider/uphold"
+	wallet "github.com/brave-intl/bat-go/wallet/service"
+	"github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+	"github.com/shopspring/decimal"
+)
+
+// Service contains datastore
+type Service struct {
+	wallet    wallet.Service
+	cbClient  cbr.Client
+	datastore Datastore
+}
+
+// InitService creates a service using the passed datastore and clients configured from the environment
+func InitService(datastore Datastore) (*Service, error) {
+	cbClient, err := cbr.New()
+	if err != nil {
+		return nil, err
+	}
+
+	walletService, err := wallet.InitService(datastore, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	service := &Service{
+		wallet:    *walletService,
+		cbClient:  cbClient,
+		datastore: datastore,
+	}
+	return service, nil
+}
+
+// CreateOrderFromRequest creates an order from the request
+func (service *Service) CreateOrderFromRequest(req CreateOrderRequest) (*Order, error) {
+	totalPrice := decimal.New(0, 0)
+	orderItems := []OrderItem{}
+	for i := 0; i < len(req.Items); i++ {
+		orderItem, err := createOrderItemFromMacaroon(req.Items[i].SKU, req.Items[i].Quantity)
+		if err != nil {
+			return nil, err
+		}
+		totalPrice = totalPrice.Add(orderItem.Subtotal)
+
+		orderItems = append(orderItems, *orderItem)
+	}
+
+	order, err := service.datastore.CreateOrder(totalPrice, "brave.com", "pending", orderItems)
+
+	return order, err
+}
+
+// UpdateOrderStatus checks to see if an order has been paid and updates it if so
+func (service *Service) UpdateOrderStatus(orderID uuid.UUID) error {
+	order, err := service.datastore.GetOrder(orderID)
+	if err != nil {
+		return err
+	}
+
+	sum, err := service.datastore.GetSumForTransactions(orderID)
+	if err != nil {
+		return err
+	}
+
+	if sum.GreaterThanOrEqual(order.TotalPrice) {
+		err = service.datastore.UpdateOrder(orderID, "paid")
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// CreateTransactionFromRequest queries the endpoints and creates a transaciton
+func (service *Service) CreateTransactionFromRequest(req CreateTransactionRequest, orderID uuid.UUID) (*Transaction, error) {
+	var wallet uphold.Wallet
+	upholdTransaction, err := wallet.GetPublicTransaction(req.ExternalTransactionID)
```

Comment:
I think for now I would use the non-public method for getting a transaction so we can check that the destination of the funds is correct ( that will work for AC at least )

Reply:
https://github.com/brave-intl/bat-go/pull/226/commits/6a990da291fd5560fb941c55e10135dbf695d937